flink-libraries/flink-table/src/main/scala/org/apache/flink/table/api/table.scala

```diff
@@ -911,7 +911,21 @@ class GroupedTable(
     */
   def select(fields: String): Table = {
     val fieldExprs = ExpressionParser.parseExpressionList(fields)
-    select(fieldExprs: _*)
+
+    //get the correct expression for UDAGGFunctionCall
+    val input: Seq[Expression] = fieldExprs.zipWithIndex.map {
+      case (Call(name, args), idx) => {
```

Comment:
This does not catch all cases. The agg function call could be nested in one or more expressions:
```
select('x, weightAvgFun('long, 'int) * 2)
```
So we need to recursively search each expression for UDAGG calls.

It would also be good to make this a utility function in `ProjectionTranslator`.

I think this needs to be added to `Table.select()` as well (non-grouped, non-windowed aggregation).